File: arrays/deletion/removeElement.java
package deletion;

class removeElement {
  public int remove(int[] nums, int val) {
    int k = 0;
    int r1 = nums.length - 1;
    for (int i = 0; i < nums.length; i++) {
      if (nums[i] == val) {
        nums[r1] = nums[i];
      }
    }
    return k;
  }
}

File: c++/Arrays/remove_element.cpp
#include <vector>
using namespace std;

class Solution
{
public:
    int removeElement(vector<int> &nums, int val)
    {
        if (nums.empty())
            return 0;

        int i = 0; // i tracking non val elements

        for (int j = 0; j < nums.size(); j++)
        {
            if (nums[j] != val)
            {
                nums[i] = nums[j];
                i++;
            }
        }

        return i;
    }
};
